src/python/deepgnn/graph_engine/snark/tests/convert_test.py

```
@@ -720,6 +775,309 @@ def test_sanity_edge_sparse_features_data(graph_with_sparse_features):
+        assert result[28:36] == (1).to_bytes(8, byteorder=sys.byteorder)
+        assert result[36:40] == (0).to_bytes(4, byteorder=sys.byteorder)
+
+
+def test_edge_list_error_checking():
+    decoder = EdgeListDecoder()
+    with pytest.raises(StopIteration):
+        next(decoder.decode(""))
+    with pytest.raises(ValueError):
+        next(decoder.decode("x"))
+    with pytest.raises(RuntimeError):
+        next(decoder.decode("0,-1"))
+    with pytest.raises(RuntimeError):
+        next(decoder.decode("0,-1,4"))
+    with pytest.raises(ValueError):
+        next(decoder.decode("0,-1,4,x"))
+    # ignores feature vector if fails to read
+    next(decoder.decode("0,-1,4,1,bad_key"))
+    next(decoder.decode("0,-1,4,1,float32"))
+    with pytest.raises(ValueError):
+        next(decoder.decode("0,-1,4,1,float32,2"))
+    with pytest.raises(ValueError):
+        next(decoder.decode("0,-1,4,1,float32,2,1"))
+    with pytest.raises(ValueError):
+        next(decoder.decode("0,-1,4,1,float32,2,1,x"))
+    next(decoder.decode("0,-1,4,1,float32,2,1,1"))
+    next(decoder.decode("0,-1,4,1,binary,1,test"))
+    next(decoder.decode("0,-1,4,1,float32,0"))
+    with pytest.raises(RuntimeError):
+        gen = decoder.decode("0,1")
+        next(gen)
+    with pytest.raises(RuntimeError):
+        gen = decoder.decode("0,1,4")
+        next(gen)
+    gen = decoder.decode("0,1,4,1")
+    next(gen)
+
+
+def test_edge_list_binary_escape():
+    decoder = EdgeListDecoder()
+    src, dst, typ, weight, features = next(
+        decoder.decode("0,-1,0,1.0,binary,1,test,feature")
+    )
+    assert features[0] == "test"
+    src, dst, typ, weight, features = next(
+        decoder.decode("0,-1,0,1.0,binary,1,test\,feature")
+    )
+    assert features[0] == "test,feature"
+    src, dst, typ, weight, features = next(
+        decoder.decode(r"0,-1,0,1.0,binary,1,test\\,feature")
+    )
+    assert features[0] == r"test\,feature"
```

This is a wrong result, there should be 2 features: `test\` and `feature`

---

In order to get 2 features it would have ot be "binary,1,test,binary,1,feature", maybe it should error out instead of silently ignoring?

---

Looks like at least 3 more test cases are needed:
1. `0,-1,0,1.0,binary,1,test\\,binary,1,hello` -> 2 features
2. `0,-1,0,1.0,binary,2,test\\,hello` -> 2 features
3. `0,-1,0,1.0,binary,1,test\\,hello` -> error

---

Second test is invalid, each feature vector needs a key and length. Binary_feature only supports length 1. I'll add the other two.

---

>Second test is invalid, each feature vector needs a key and length. 

You still need to test a combination of multiple binary features:
* `0,-1,0,1.0,binary,1,test\,hello,binary,1,test`
* `0,-1,0,1.0,binary,1,test\,hello\\,binary,1,test`
and so on..